src/Microsoft.Extensions.ServiceDiscovery.Yarp/ServiceDiscoveryDestinationResolver.cs

```diff
@@ -118,4 +137,22 @@ private string GetDefaultScheme(Uri originalUri)
             return originalUri.Scheme;
         }
     }
+
+    private static bool IsLocalhost(EndPoint endpoint)
```

Comment:
Can this ever be `UriEndpoint("http://127.0.0.1")`

Reply:
Fixed using `Uri.IsLoopback`. I added more test coverage to ensure we treat the following as localhost:
* 127.0.0.1
* localhost
* ::1